support/cas-server-support-saml-idp-metadata/src/main/java/org/apereo/cas/support/saml/services/idp/metadata/cache/resolver/JsonResourceMetadataResolver.java

```diff
@@ -60,7 +61,10 @@ public JsonResourceMetadataResolver(final SamlIdPProperties samlIdPProperties,
             this.metadataTemplate = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
             val md = samlIdPProperties.getMetadata();
             val location = SpringExpressionLanguageValueResolver.getInstance().resolve(md.getLocation());
-            this.jsonResource = new FileSystemResource(new File(location, "saml-sp-metadata.json"));
+            val normalizedLocation = location.startsWith("file:") ? location : "file:" + location;
+            val resourceLoader = new DefaultResourceLoader();
+            val metadataDir = resourceLoader.getResource(normalizedLocation).getFile();
+            this.jsonResource = new FileSystemResource(new File(metadataDir, "saml-sp-metadata.json"));
```

Comment:
This should follow the same approach as https://github.com/apereo/cas/blob/master/support/cas-server-support-saml-idp-metadata/src/main/java/org/apereo/cas/support/saml/services/idp/metadata/cache/resolver/UrlResourceMetadataResolver.java#L67-L69